File: selection-sort/Selection.js
function selection(arr) {
    var _a;
    var min;
    for (var i = 0; i < arr.length - 1; i++) {
        min = i;
        for (var j = i + 1; j < arr.length; j++) {
            if (arr[j] < arr[min]) {
                min = j;
            }
        }
        if (min !== i)
            _a = [arr[min], arr[i]], arr[i] = _a[0], arr[min] = _a[1]; // swap values
    }
    return arr;
}
console.log(selection([12, 31, 25, 8, 32, 17]));
console.log(selection([80, 10, 29, 11, 8, 30, 15]));

File: selection-sort/Selection.py
import time


def selection(arr):
    for i in range(0, len(arr) - 1):
        min = i
        for j in range(i + 1, len(arr)):
            if arr[j] < arr[min]:
                min = j
        if min != i:
            arr[i], arr[min] = arr[min], arr[i]  # swap values
    return arr


start = time.time()
arr = selection([12, 31, 25, 8, 32, 17])
end = time.time()
print(arr, "time: " + str((end - start) * 10**3))
